test/internal_rules.jl

```diff
@@ -379,12 +312,91 @@ end
 
         A = [1.3 0.5; 0.5 1.5]
         b = [1., 2.]
-        V = [1.0 0.0; 0.0 0.0]
         dA = zero(A)
         Enzyme.autodiff(Reverse, h, Active, Duplicated(A, dA), Const(b))
 
         dA_sym = - (transpose(A) \ [1.0, 0.0]) * transpose(A \ b)
-        @test isapprox(dA, dA_sym)
+        @test isapprox((dA + dA') / 2, (dA_sym + dA_sym') / 2)
+    end
+    @testset "Unit test for `cholesky` (regression test for #1307)" begin
+        # This test checks the `cholesky` rules without involving `ldiv!`
+        function f(A)
+            C = cholesky(A * adjoint(A))
+            return sum(abs2, C.L * C.U)
+        end
+        @testset for TE in (Float64, ComplexF64)
+            A = rand(TE, 3, 3)
```

Comment:
Let's make this a little harder.
```suggestion
            A = rand(TE, 5, 5)
```